test/rack/session/redis_test.rb

```diff
@@ -347,6 +347,22 @@
     end
   end
 
+  describe "on a connection failure" do
+    it "#get_session returns the default when Redis::CannotConnectError is raised" do
+      redis_store = Rack::Session::Redis.new("redis://127.0.0.1:6380/1")
```

Comment:
instead of using a mock, why not just try to connect to a bogus redis URL like `redis://definitelynot.redis:1337`